src/sinks/aws_cloudwatch_logs/request.rs

```diff
@@ -212,6 +233,19 @@ impl Future for CloudwatchFuture {
                         return Poll::Ready(Ok(()));
                     }
                 }
+
+                State::PutRetentionPolicy(fut) => {
+                    match ready!(fut.poll_unpin(cx)) {
+                        Ok(__) => {}
+                        Err(error) => {
+                            return Poll::Ready(Err(CloudwatchError::PutRetentionPolicy(error)))
+                        }
+                    }
+
+                    info!(message = "Retention policy updated for stream.", name = %self.client.stream_name);
+
+                    self.state = State::DescribeStream(self.client.describe_stream());
```

Comment:
I think this should be `CreateStream`, in `CreateGroup` that is the state it moves to if we don't have `retention_enabled`.

```suggestion
                 self.state = State::CreateStream(self.client.create_log_stream());
```

The code as it stands would still work if `create_stream` is true because `DescribeStream` would create the stream if it doesn't exist. But if false, then we would get an error whereas before we wouldn't.